### ISSUE TITLE
Security considerations about Pod user

### ISSUE THREAD

**Issue by gillg** [PrivateBin/helm-chart#31]
Hi,

When you enforce a "good" PSP to deny launch root pods you can't launch privatebin pod.
In fact, container runs as a random non root user (called "nobody" and created in dockerfile), but kubernetes can't determine its UID to check if is different of 0.

`Error: container has runAsNonRoot and image has non-numeric user (nobody), cannot verify user is non-root`

I made a little test like this :
```yaml
    spec:
      securityContext:
        runAsUser: 65534
      containers:
        - name: {{ .Chart.Name }}
          image: "{{ .Values.image.repository }}:{{ default .Chart.AppVersion .Values.image.tag }}"
          imagePullPolicy: {{ .Values.image.pullPolicy }}
```

It works perfectly because 65534 is the UID of "nobody" user, but it seems not a stable solution because UID is not guaranteed.
Anyone has any idea to make it clean ?
Or we just should make a change in dockerfile to fix an ID ? https://github.com/PrivateBin/docker-nginx-fpm-alpine/blob/master/Dockerfile#L54

**Comment by elrido** [comment]
We probably could avoid the issue by using numeric IDs (`65534:82`) in this line:
https://github.com/PrivateBin/docker-nginx-fpm-alpine/blob/4b98c98f4927e6dbef8222cc9ef25fb6b7b87870/Dockerfile#L63

**Comment by gillg** [comment]
I think too, but if uid is not set into useradd command is not 100% guaranteed :/
Maintainers are same for both projects ?
Who can take this decision ?

Regards

**Comment by elrido** [comment]
The command `adduser nobody www-data` adds the user to the group. nobody is predefined on that ID in alpine 3.12 and the www-data user with GID 82 gets created by the nginx package.

Yes, I can change the IDs there this evening and the `latest` tag produced tonight should then contain this.

**Comment by gillg** [comment]
Super efficient 🎉 !

**Comment by elrido** [comment]
So that change got done over in the container repo - the [scheduled build](https://github.com/PrivateBin/docker-nginx-fpm-alpine/actions?query=workflow%3A%22Deploy+multi-architecture+Docker+images+for+privatebin+with+buildx%22) should push the update to the nightly tag some time after midnight UTC:
https://github.com/PrivateBin/docker-nginx-fpm-alpine/commit/7994a768ad5d9df576b4edbaed6bcb37bd20c5a6

Follow up question: In the containers README we actually [suggest some security context settings for k8s](https://github.com/PrivateBin/docker-nginx-fpm-alpine#kubernetes-deployment) - I am running it that way on a k3s cluster. Would it cause any upgrade issues for helm users, if we added these settings in the helm chart as well? Would it for example conflict with having runAsNonRoot being enforced? I can raise a merge request for it.

**Comment by gillg** [comment]
In my opinion it's a good thing to report these advices in helm chart.
Even in a very restrictive PSP pod should be created. And in every cases, actual chart not work  for a RunAsNonRoot condition...

If you want, I could test it for you ;)

**Comment by gillg** [comment]
I updated my chart today, and it works without forcing `securityContext` in helm !
K8s detect the real UID and accept to launch pod with my PSP restriction :)

We could force securityContext in helm to be more explicit, but it's not mandatory.

**Comment by bdashrad** [comment]
Should be resolved by #32
